src/Sarif.UnitTests/Converters/AndroidStudioConverterTests.cs

```diff
@@ -137,238 +139,291 @@ public void AndroidStudioConverter_ConvertToSarifIssue_HintsAreStapledToDescript
         }
 
         [TestMethod]
-        public void AndroidStudioConverter_ConvertToSarifIssue_AttributeKeyIsPersistedInProperties()
+        public void AndroidStudioConverter_ConvertToSarifResult_AttributeKeyIsPersistedInProperties()
         {
             var builder = AndroidStudioProblemTests.GetDefaultProblemBuilder();
             builder.AttributeKey = "key";
             var uut = new AndroidStudioProblem(builder);
-            Result result = AndroidStudioConverter.ConvertProblemToSarifIssue(uut);
+            Result result = new AndroidStudioConverter().ConvertProblemToSarifResult(uut);
             result.Properties.Should().Equal(new Dictionary<string, string> {
                 {"attributeKey", "key"}
             });
         }
 
         [TestMethod]
-        public void AndroidStudioConverter_ConvertToSarifIssue_SeverityIsPersistedInProperties()
+        public void AndroidStudioConverter_ConvertToSarifResult_SeverityIsPersistedInProperties()
         {
             var builder = AndroidStudioProblemTests.GetDefaultProblemBuilder();
             builder.Severity = "warning";
             var uut = new AndroidStudioProblem(builder);
-            Result result = AndroidStudioConverter.ConvertProblemToSarifIssue(uut);
+            Result result = new AndroidStudioConverter().ConvertProblemToSarifResult(uut);
             result.Properties.Should().Equal(new Dictionary<string, string> {
                 {"severity", "warning"}
             });
         }
 
         [TestMethod]
-        public void AndroidStudioConverter_ConvertToSarifIssue_MultiplePropertiesArePersisted()
+        public void AndroidStudioConverter_ConvertToSarifResult_MultiplePropertiesArePersisted()
         {
             var builder = AndroidStudioProblemTests.GetDefaultProblemBuilder();
             builder.AttributeKey = "key";
             builder.Severity = "warning";
             var uut = new AndroidStudioProblem(builder);
-            Result result = AndroidStudioConverter.ConvertProblemToSarifIssue(uut);
+            Result result = new AndroidStudioConverter().ConvertProblemToSarifResult(uut);
             result.Properties.Should().Equal(new Dictionary<string, string> {
                 {"severity", "warning"},
                 {"attributeKey", "key"}
             });
         }
 
         [TestMethod]
-        public void AndroidStudioConverter_ConvertSarifIssue_RecordsTopLevelFileAsSourceFile()
+        public void AndroidStudioConverter_ConvertSarifResult_RecordsTopLevelFileAsSourceFile()
         {
             var builder = AndroidStudioProblemTests.GetDefaultProblemBuilder();
             builder.File = "expected_file.java";
             builder.EntryPointType = "file";
             builder.EntryPointName = "bad_file.java";
-            Location loc = GetLocationForBuilder(builder);
-            Assert.AreEqual("expected_file.java", loc.ResultFile.Uri.ToString());
+            Location loc = GetLocationInfoForBuilder(builder).Location;
```

Comment:
You'll see later that when we process a `builder`, we create not only a `Location` object, but an array of `LogicalLocationComponent` objects (because the logical location components are no longer part of the `Location`). The renamed `GetLocationInfoForBuilder` class now returns a `LocationInfo` object that contains those two things.